wpical/src/main/native/include/fieldmap.h

```diff
@@ -0,0 +1,60 @@
+// Copyright (c) FIRST and other WPILib contributors.
+// Open Source Software; you can modify and/or share it under the terms of
+// the WPILib BSD license file in the root directory of this project.
+
+#pragma once
+
+#include <numbers>
+#include <vector>
+
+#include <Eigen/Dense>
+#include <Eigen/Geometry>
+#include <tagpose.h>
+#include <wpi/json.h>
+
+class fieldmap {
+ public:
+  fieldmap() = default;
+  explicit fieldmap(const wpi::json& json) {
+    double field_length_meters =
+        static_cast<double>(json.at("field").at("length"));
+    double field_width_meters =
+        static_cast<double>(json.at("field").at("width"));
+    for (const auto& tag : json.at("tags").items()) {
+      double tagXPos =
+          static_cast<double>(tag.value().at("pose").at("translation").at("x"));
+      double tagYPos =
+          static_cast<double>(tag.value().at("pose").at("translation").at("y"));
+      double tagZPos =
+          static_cast<double>(tag.value().at("pose").at("translation").at("z"));
+      double tagWQuat = static_cast<double>(
+          tag.value().at("pose").at("rotation").at("quaternion").at("W"));
+      double tagXQuat = static_cast<double>(
+          tag.value().at("pose").at("rotation").at("quaternion").at("X"));
+      double tagYQuat = static_cast<double>(
+          tag.value().at("pose").at("rotation").at("quaternion").at("Y"));
+      double tagZQuat = static_cast<double>(
+          tag.value().at("pose").at("rotation").at("quaternion").at("Z"));
+
+      tagVec.emplace_back(tagXPos, tagYPos, tagZPos, tagWQuat, tagXQuat,
+                          tagYQuat, tagZQuat, field_length_meters,
+                          field_width_meters);
+    }
+  }
+
+  const tag::Pose& getTag(size_t tag) const { return tagVec[tag - 1]; }
+
+  int getNumTags() const { return tagVec.size(); }
+
+  static double minimizeAngle(double angle) {
```

Comment:
what happens if I have an angle > 720? this function wont work right?

Reply:
I realized I used radians in the commit message, but this should work better right?